### PR TITLE
Documentation rework

### PR DESCRIPTION
**Pull request by pawelbudziszewski**
- Moved README contents to "Model description" and extended
- Added Step-by-step tutorial
- Added description of the example model (#2)
- Created a new README
- Updated all contents to the latest changes

### CODE REVIEW

File: Docs/Tutorial.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+For each wheel entity, add `Wheel Controller` component. Set the parameters as follows:
+
+1. Set `Wheel radius` to match the radius of the wheel.
+2. Set `Spring rest length` to the distance from the top of the suspension to the center of the wheel. Typically this 
+can be adjusted to match the visual model.
+3. Set `Spring travel` according to your needs. This is the maximum distance the wheel can travel up and down.
+4. Set `Wheel visual entity` to the entity containing visual.
+
+The wheel component provides a convenient visualizatio, which can help setting these parameters.
```
Comment (c00kiepreferences) [inline Docs/Tutorial.md]:
```suggestion
The wheel component provides a convenient visualization, which can help setting these parameters.
```

File: README.md [hunk 2]
```diff
@@ -1,78 +1,32 @@
...
 
-#### Inputs
+You can follow the [Step-by-step tutorial](Docs/Tutorial.md) to learn how to set up the model.
 
-- `Acceleration` - acceleration in m/s^2 to be applied to the vehicle
-- `Steering angle` - steering angle in degrees
+### Example car
 
-#### Parameters
+The example car prefab is located in the [Assets/ExampleCar](Assets/ExampleCar) folder. It contains a prefab with
```
Comment (c00kiepreferences) [inline README.md]:
```suggestion
The example car prefab is located in the [Assets/ExampleCar](./Assets/ExampleCar) folder. It contains a prefab with
```

File: README.md [hunk 3]
```diff
@@ -1,78 +1,32 @@
...
-Where:
-
-- **_fc_** - lateral friction coefficient
-- **_v_** - lateral velocity
-- **_c1_** - lateral friction C1 coefficient, is responsible for coefficient scaling - the higher _c1_ value, the larger lateral force.
-- **_c2_** - lateral friction C2 coefficient, indicates how steep the declining part of the curve is - the higher _c2_ value, the lower values of the resultant coefficient in larger velocity. _c2_ should be larger than 1.0. 
-- **_c3_** - lateral friction C3 coefficient, is responsible for velocity scaling - the higher _c3_ value, the faster max value is reached.
+![Example car](Docs/Images/Example_car1.png)
+![Example car](Docs/Images/Example_car1.gif)
+![Example car](Docs/Images/Example_car2.gif)
```
Comment (c00kiepreferences) [inline README.md]:
```suggestion
![Example car](./Docs/Images/Example_car1.png)
![Example car](./Docs/Images/Example_car1.gif)
![Example car](./Docs/Images/Example_car2.gif)
```

File: Docs/Model_description.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+The complete vehicle asset should consist of:
+- Entity with `Rigid Body` - physics body of the vehicle, with mass, inertia, and collision geometry, as well as damping
+- Chassis visualization - visual representation of the vehicle. It must be in the same entity, or a child of the rigid
+  body entity.
+- `Vehicle Controller` component.
+- Four entities with `Wheel Controller` components, children of the rigid body entity.
+- Visualization of the wheels, children of the wheel controller entities.
+- Optionally, entity with visualization of the steering wheel, child of the rigid body entity.
+
+![Vehicle dynamics diagram](Images/Vehicle_model_entities.png)
```
Comment (c00kiepreferences) [inline Docs/Model_description.md]:
```suggestion
![Vehicle dynamics diagram](./Images/Vehicle_model_entities.png)
```

File: Docs/Model_description.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    - Driving torque is not calculated, so driving force is estimated based on the acceleration coefficient.
+    - Wheel don't have mass or inertia.
+- Engine and transmission are not simulated. The vehicle is controlled by a force derived from acceleration coefficient,
+  and divided into wheels based on ground force.
+
+## Components
+
+### Vehicle Controller component
+
+![Vehicle Controller Component](Images/Vehicle_controller_component.png)
```
Comment (c00kiepreferences) [inline Docs/Model_description.md]:
```suggestion
![Vehicle Controller Component](./Images/Vehicle_controller_component.png)
```

File: README.md [hunk 6]
```diff
@@ -1,78 +1,32 @@
...
+point of contact with the ground.
 
-### Simple Vehicle Controller component
+## Getting started
 
-![Simple_vehicle_controller_component.png](Docs/Images/Simple_vehicle_controller_component.png)
+### Model Description
 
-#### Description
+If you plan to use this model, be sure to read [Model Description](Docs/Model_description.md) to:
```
Comment (c00kiepreferences) [inline README.md]:
```suggestion
If you plan to use this model, be sure to read [Model Description](./Docs/Model_description.md) to:
```

File: README.md [hunk 7]
```diff
@@ -1,78 +1,32 @@
...
+- get familiar with the model structure and functionality,
+- see the requirements and limitations of the model,
+- learn about components and their parameters.
 
-This component is used to control the vehicle model. It must be placed in the same entity as the vehicle's rigid body.
+### Tutorial
 
-#### Inputs
+You can follow the [Step-by-step tutorial](Docs/Tutorial.md) to learn how to set up the model.
```
Comment (c00kiepreferences) [inline README.md]:
```suggestion
You can follow the [Step-by-step tutorial](Docs/Tutorial.md) to learn how to set up the model.
```
```suggestion
You can follow the [Step-by-step tutorial](./Docs/Tutorial.md) to learn how to set up the model.
```

File: Docs/Model_description.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+- `Wheels of the front axle` - list of wheels (entities with `Wheel Controller`) of the front axle, ordered from left to right
+- `Wheels of the rear axle` - list of wheels (entities with `Wheel Controller`) of the rear axle, ordered from left to right
+- `Front wheels drive` - flag indicating if the front wheels are driving
+- `Rear wheels drive` - flag indicating if the rear wheels are driving
+- `Enable manual control` - flag indicating if the vehicle can be controlled using arrow keys
+- `Steering wheel entity` - entity with a steering wheel visualization, which will be rotated according to the steering angle
+
+### Wheel Controller component
+
+![Wheel Controller component](Images/Wheel_controller_component.png)
```
Comment (c00kiepreferences) [inline Docs/Model_description.md]:
```suggestion
![Wheel Controller component](./Images/Wheel_controller_component.png)
```

File: Docs/Model_description.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+Lateral wheel force is proportional to:
+
+- Ground force
+- Friction coefficient
+- Longitudinal velocity (inverse proportionality)
+
+The friction coefficient is calculated based on lateral velocity. The formula is as follows:
+
+![Lateral Friction Coefficient](Images/lateral_friction_coeff.png)
```
Comment (c00kiepreferences) [inline Docs/Model_description.md]:
```suggestion
![Lateral Friction Coefficient](./Images/lateral_friction_coeff.png)
```

File: Docs/Tutorial.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+
+![Tutorial](Images/Tutorial_01.png)
+
+Origin and orientation of the `Wheel FR` entity:
+
+![Tutorial](Images/Tutorial_02.png)
+
+Origin and orientation of the `Wheel FR visual` entity:
+
+![Tutorial](Images/Tutorial_03.png)
```
Comment (c00kiepreferences) [inline Docs/Tutorial.md]:
```suggestion
![Tutorial](./Images/Tutorial_01.png)

Origin and orientation of the `Wheel FR` entity:

![Tutorial](./Images/Tutorial_02.png)

Origin and orientation of the `Wheel FR visual` entity:

![Tutorial](./Images/Tutorial_03.png)
```

File: Docs/Tutorial.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+
+1. Set `Wheel radius` to match the radius of the wheel.
+2. Set `Spring rest length` to the distance from the top of the suspension to the center of the wheel. Typically this 
+can be adjusted to match the visual model.
+3. Set `Spring travel` according to your needs. This is the maximum distance the wheel can travel up and down.
+4. Set `Wheel visual entity` to the entity containing visual.
+
+The wheel component provides a convenient visualizatio, which can help setting these parameters.
+
+![Tutorial](Images/Tutorial_04.png)
```
Comment (c00kiepreferences) [inline Docs/Tutorial.md]:
```suggestion
![Tutorial](./Images/Tutorial_04.png)
```

File: Docs/Tutorial.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+
+The rest of component's parameters should be set experimentally. In the case of typical sport car, you can use following 
+values as a starting point:
+
+![Tutorial](Images/Wheel_controller_component.png)
```
Comment (c00kiepreferences) [inline Docs/Tutorial.md]:
```suggestion
![Tutorial](./Images/Wheel_controller_component.png)
```

File: Docs/Tutorial.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+Add `Vehicle Controller` component to the `Vehicle` entity:
+1. Add `Wheel` entities to the `Wheels` array. Remember to maintain the order of the wheels from left to right. 
+2. Set the `Wheelbase` and `Track` parameters to match your model.
+3. Set other parameters according to your needs.
+4. Turn on `Manual control` if you want to control the vehicle manually.
+5. Set `Steering wheel entity` if you have the steering wheel visualization.
+
+The component should look similar to this:
+
+![Tutorial](Images/Vehicle_controller_component.png)
```
Comment (c00kiepreferences) [inline Docs/Tutorial.md]:
```suggestion
![Tutorial](./Images/Vehicle_controller_component.png)
```

**Review by c00kiepreferences — APPROVED** [review]
Looks great to me, readme seems to be clear, approachable and comprehensive.

**Comment by jhanca-robotecai** [comment]
I pushed some changes to this branch so that I did not have to push it via code review. @c00kiepreferences please review my changes. Thanks.

**Review by c00kiepreferences — APPROVED** [review]

**Pull request merged**